sql/sql_priv.h

```diff
@@ -354,6 +354,7 @@
   data dictionary without changing table rows
 */
 #define IS_EQUAL_PACK_LENGTH 2
+#define IS_EQUAL_BUT_COLLATE 3
```

Comment:
In https://github.com/MariaDB/server/pull/1251/files#r271278409 I suggested adding a `/** comment */`. If you don’t think that these values can be used as bitmasks, then I would suggest replacing the `#define` with `enum is_equal_t` that would be defined inside `class Field`, and using the `Field::is_equal_t` wherever appropriate.

Reply:
I don't this this can be a bit mask. As I understand originally `Field::is_equal()` returned `bool`. Then the became two degrees of inequality: total inequality and inequality which allows instant enlarging. This patch adds another inequality degree: everything is equal, but not `COLLATE`. I don't think we can combine two 'soft' inequalities. At least for now.

As of introducing and `enum` here, I totally agree it's a good thing. But probably it's not time to do in in a feature patch. I'd create a separate PR for it.